src/radical/pilot/agent/scheduler/continuous.py

```diff
@@ -202,11 +209,17 @@ def _alloc_node(self, node, requested_cores, requested_gpus,
 
         # we can serve the partial or full request - alloc the chunks we need
         # FIXME: chunk gpus, too?
+
+        # MING: We don't really need to include chunk in the arithmetic if
+        #       they cancel out right?
         alloc_cores = min(requested_cores, free_cores) / chunk * chunk
         alloc_gpus  = min(requested_gpus , free_gpus )
 
         # now dig out the core and gpu IDs.
         for idx,state in enumerate(node['cores']):
+            
+            # MING: Just to check, is this true only when there are no free
+            #       cores?
```

Comment:
Hmm, no, that should break when no more cores are _needed_.  Or am I missing the point of your question?

Reply:
Oh I see. I get it now. I was just confused about when the loop would break